garden-backend-service/src/api/routes/users.py

```diff
@@ -62,71 +63,124 @@ async def update_user_info(
     return user_response
 
 
-@router.get("/gardens/saved")
-async def get_saved_garden_dois(
-    authed_user: User = Depends(authed_user),
+@router.get("/{user_uuid}/saved/gardens", response_model=list[GardenMetadataResponse])
+async def get_saved_gardens(
+    user_uuid: UUID,
     db: AsyncSession = Depends(get_db_session),
-) -> list[str]:
-    return await _get_saved_garden_dois(authed_user, db)
+) -> list[GardenMetadataResponse]:
+    """Fetch the users list of saved gardens."""
+    user: User | None = await User.get(db, identity_id=user_uuid)
 
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"No user with uuid {user_uuid} found.",
+        )
 
-@router.patch("/gardens/saved/{doi:path}")
+    # We need to explicitly load the saved_gardens field
+    await db.refresh(user, ["saved_gardens"])
+    return [GardenMetadataResponse.from_orm(garden) for garden in user.saved_gardens]
```

Comment:
I also like pydantic v1's `from_orm` name better for this method, but for the sake of consistency/not getting a bunch of deprecated warnings let's use their preferred method name
```suggestion
    return [GardenMetadataResponse.model_validate(garden) for garden in user.saved_gardens]
```

Reply:
though since we specified a response_model, fastapi would also do this conversion automatically:

```suggestion
    return user.saved_gardens
```